dialer.go

```diff
@@ -374,11 +197,138 @@ func (d *customDialer) writeWARCFromConnection(reqPipe, respPipe *io.PipeReader,
 	return nil
 }
 
-func newCustomDialer(httpClient *CustomHTTPClient) *customDialer {
-	var d = new(customDialer)
+func (d *customDialer) readResponse(respPipe *io.PipeReader, warcTargetURIChannel chan string, recordChan chan *Record) error {
+	// Initialize the response record
+	var responseRecord = NewRecord()
+	responseRecord.Header.Set("WARC-Type", "response")
+	responseRecord.Header.Set("Content-Type", "application/http; msgtype=response")
 
-	d.Timeout = 5 * time.Second
-	d.client = httpClient
+	// Read 2MB to memory and if there's more to be read move it to a file
+	_, err := io.Copy(responseRecord.Content, respPipe)
+	if err != io.EOF && err != nil {
```

Comment:
https://pkg.go.dev/io#Copy

"A successful Copy returns err == nil, not err == EOF. Because Copy is defined to read from src until EOF, it does not treat an EOF from Read as an error to be reported."

Reply:
Indeed! Thanks.